apps/bank/obuk_utils.go

```diff
@@ -0,0 +1,168 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/cloudentity/openbanking-quickstart/openbanking/obuk/accountinformation/models"
+	paymentModels "github.com/cloudentity/openbanking-quickstart/openbanking/obuk/paymentinitiation/models"
+	"github.com/go-openapi/strfmt"
+
+	acpClient "github.com/cloudentity/acp-client-go/models"
+)
+
+func OBUKMapError(err error) (int, models.OBError1) {
+	msg := err.Error()
+	resp := models.OBError1{
+		Message: &msg,
+	}
+
+	if errors.As(err, &errNotFound) {
```

Comment:
this is really complicated solution to map a http status code to an error. Maybe do sth like this instead

```
type Error struct {
  Code int `json:"code"`
  Message string `json:"message"`
  Error error `json:"-"`
}

func (e *Error) WithError(err error) *Error {
  return &Error{
    Code: e.Code,
    Message: e.Message,
    Err: errors.WithStack(err),
  }
}

func (e *Error) Unwrap() error {
  return e.Err
}

func NewError(err error) *Error {
  var e Error

  if errors.As(err, &e) {
    return e
  }

  return &Error{Code: http.InternalServerError, Message: "unknown error", Err: error}
}

var (
  ErrNotFound = &Error{
    Code: http.StatusNotFound,
    Message: "entity not found",
  }
)
```

look at https://github.com/cloudentity/go-tools/blob/master/httpx/error.go for more inspiration (but you don't have to copy everything 1:1)

also read https://blog.golang.org/go1.13-errors